src/main/java/de/qabel/desktop/crashReports/HockeyApp.java

```diff
@@ -1,77 +1,83 @@
 package de.qabel.desktop.crashReports;
 
+
+import de.qabel.desktop.hockeyapp.HockeyAppConfiguration;
+import de.qabel.desktop.hockeyapp.HockeyAppVersion;
+import de.qabel.desktop.hockeyapp.VersionClient;
 import org.apache.http.HttpEntity;
 import org.apache.http.NameValuePair;
 import org.apache.http.client.HttpClient;
 import org.apache.http.client.entity.UrlEncodedFormEntity;
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.entity.mime.MultipartEntityBuilder;
 import org.apache.http.entity.mime.content.ByteArrayBody;
-import org.apache.http.impl.client.HttpClients;
 import org.apache.http.message.BasicNameValuePair;
 import org.apache.http.protocol.HTTP;
+import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 
 public class HockeyApp implements CrashReportHandler {
-    private static final String APP_ID = "3b119dc227334d2d924e4e134c72aadc";
-    private static final String TOKEN = "350b097ef0964b17a0f3907050de309d";
-
-    private HttpClient httpClient = HttpClients.createMinimal();
 
-    @Override
-    public void sendFeedback(String feedback, String name, String email) throws IOException {
+    HockeyAppConfiguration config;
+    VersionClient versionClient;
+    HttpClient httpClient;
 
-        URI uri;
-        try {
-            uri = new URI("https://rink.hockeyapp.net/api/2/apps/" + APP_ID + "/feedback");
-        } catch (URISyntaxException e) {
-            throw new IllegalStateException("invalid App Id");
-        }
-        HttpPost httpPost = new HttpPost(uri);
-        httpPost.addHeader("X-HockeyAppToken", TOKEN);
+    public HockeyApp(String currentVersion, HttpClient httpClient) {
+        this.config = new HockeyAppConfiguration(currentVersion, httpClient);
+        this.httpClient = httpClient;
+        this.versionClient = new VersionClient(config, httpClient);
```

Comment:
The overall distribution of methods between your classes seems more random than by purpose. You extracted the Version part of the HockeyAppApi to the VersionClient, ok, but why is there no FeedbackClient for the feedback part of the api? It's in here... which would fit if there was no VersionClient, but it looks inconsistent now.

The purpose of the `HockeyAppConfiguration` is unclear. Does it encapsulate the auth stuff from the content biased request logic of the HockeyApp(Class) and it's clients? Then it should be injected and not build by it. Those constants in the HockeyAppConfiguration might end up in a config file like the `launch.properties` and then it needs to be build elsewhere and injected into the HockeyApp. So just make the HockeyAppConfiguration injectable via constructor.  ... When you do that, you will notice that you don't need to inject the currentVersion and the httpClient any longer to use the config, but only to use the versionClient.  .... when the HockeyAppConfiguration is used for building the http request objects then it should be called like `HockeyAppRequestBuilder`. The VersionClient could use that class too and you wouldn't have to use the `httpClient` any more for the VersionClient. ...so try to group your methods in classes with a clear purpose and the code will be better to understand and easier to maintain.